### DISCUSSION TITLE
Version 1.12.0

### DISCUSSION THREAD

**Discussion by trask** [open-telemetry/opentelemetry-java-instrumentation#5558] [Q&A]
Note that all artifacts other than `io.opentelemetry.javaagent:opentelemetry-javaagent` have the `-alpha` suffix attached to their version number, reflecting that they are still alpha quality and will continue to have breaking changes. Please see the [VERSIONING.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/VERSIONING.md#opentelemetry-java-instrumentation-versioning) for more details.

### 🌟 New javaagent instrumentation

- Add Azure SDK instrumentation (#5467)

### 📈 Enhancements

- Use repository interface name in spring data operation name (#5352)
- Change the way Micrometer LongTaskTimer is bridged (#5338)
- Updates to http.server_name (#5369)
- Support forwarded proto field and x-forwarded-proto (#5357)
- Remove server span peer name (#5404)
- Add peer service support back to couchbase26 (#5451)
- Remove duplicative JAXRS HttpClient instrumentation (#5430)
- Add experimental thread attributes for logs (#5474)
- Add log4j 1.2 appender MDC capture (#5475)
- Capture logback formatted message (#5497)
- Add JBoss java.util.logging support (#5498)
- Disable the messaging receive span telemetry by default (#5500)
- Remove old experimental properties in CapturedHttpHeaders (#5524)
- Avoid conflicts in Micrometer description mapping (#5452)

### 🛠️ Bug fixes

- Fix another reverse lookup (#5393)
- Regression in loading the prometheus exporter (#5408)
- Fix muzzle failure on calls to primitive array clone (#5405)
- Fix regression in spring-scheduling span name (#5436)
- Fix android desugaring for HashMap.forEach (#5468)
- Fix duplicate class definition of ContextDataProvider (#5528)
- Do not propagate gRPC deadline when propagating OTel context via javaagent (#5543)

### 🧰 Tooling

- Split out CodeAttributesGetter (#5342)
- Add prometheus smoke test (#5417)
- Set custom gRPC client/server span name extractor (#5244)
- Split out DbClientAttributesGetter and SqlClientAttributesGetter (#5354)
- Add builders for setting optional attributes on HTTP extractors (#5347)
- Write http server tests in java (#5501)
- Deprecate CapturedHttpHeaders and replace it with builder methods (#5533)
- Update to Groovy 4 (#5532)

### 🙇 Thank you
This release was possible thanks to the following contributors who shared their brilliant ideas and awesome pull requests:

@amitgud-doordash
@anuraaga
@arminru
@BodySplash
@chalin
@domyway
@eugeniyk
@fstab
@iNikem
@irizzant
@kdombeck
@landrunner
@laurit
@lmolkova
@mateuszrzeszutek
@pleeplop
@ryandens
@Stephan202
@trask
@udaysrinivasan

<hr /><em>This discussion was created from the release <a href='https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.12.0'>Version 1.12.0</a>.</em>

**Comment by Stephan202** [comment]
> Note that all artifacts other than have the suffix attached to their version number

That sentence seems to be missing two monospace words: 

> Note that all artifacts other than `io.opentelemetry.javaagent:opentelemetry-javaagent` have the `-alpha` suffix attached to their version number

(Congrats on the release!)

**Reply by trask** [reply]
oh, thx! fixed, and sent #5561 to fix the automation behind that
